Docs recipes styled components (#15093)
gatsbyjs/gatsby: docs/docs/recipes.md

- Add it to your site’s gatsby-config.js

#### Directions

1. Open the index page component (src/pages/index.js), import the styled-components package, and create some styled components

```jsx:title=src/pages/index.js
import React from "react"
import styled from "styled-components"

const Container = styled.div`
  margin: 3rem auto;
  max-width: 600px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

const Avatar = styled.img`
  flex: 0 0 96px;
  width: 96px;
  height: 96px;
  margin: 0;
`

const Username = styled.h2`
  margin: 0 0 12px 0;
  padding: 0;
`

const User = props => (
  <UserWrapper>
    <Avatar src={props.avatar} alt={props.username} />
    <Username>{props.username}</Username>
  </UserWrapper>
)

export default () => (
  <Container>
    <h1>About Styled Components</h1>
    <p>Styled Components is cool</p>
    <User
      username="Jane Doe"
      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/adellecharles/128.jpg"
    />
    <User
      username="Bob Smith"
      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/vladarbatov/128.jpg"
    />
  </Container>
)
```

> 2. Run `gatsby develop` to see the changes

Reviewer: ```suggestion
4. Run `gatsby develop` to see the changes
```

- Use the CSS-in-JS library [Styled Components](/docs/styled-components/)
- Use [CSS Modules](/tutorial/part-two/#css-modules)

### Using Styled Components

> #### Requirements

Reviewer 2: ```suggestion
#### Prerequisites
```

This PR: #15236 changed the name from Requirements to Prerequisites so probably worth updating.

- Install the necessary dependencies for styled-components, including the Gatsby plugin

> ```shell

Reviewer 2: This section actually doesn't need to include the steps to install or setup prerequisites so they don't have to repeat what's explained in a lot of other guides already 🙂 (explained [here](https://www.gatsbyjs.org/contributing/docs-templates/#recipes) a little better)

Just a couple bullet points of what should already be setup is good enough.

Author: Okay. Should I mention the packages that are needed to install?

Reviewer 2: Yeah since there are multiple it might be good to mention each of them or say something like:

```
- [Install styled-components and its necessary dependencies](/packages/gatsby-plugin-styled-components/) for use in Gatsby
```

to link to the installation steps. The idea is that the recipes just hit how to use X thing, because installation and setup is covered in README's and getting started places elsewhere.

npm install --save gatsby-plugin-styled-components styled-components babel-plugin-styled-components
```

- Add it to your site’s gatsby-config.js

```javascript:title=gatsby-config.js
module.exports = {
  plugins: [`gatsby-plugin-styled-components`],
}
```

#### Directions

1. Open the index page component (src/pages/index.js), import the styled-components package, and create some styled components

```jsx:title=src/pages/index.js
import React from "react"
import styled from "styled-components"

const Container = styled.div`
  margin: 3rem auto;
  max-width: 600px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

const Avatar = styled.img`
  flex: 0 0 96px;
  width: 96px;
  height: 96px;
  margin: 0;
`

const Username = styled.h2`
  margin: 0 0 12px 0;
  padding: 0;
`

const User = props => (
  <UserWrapper>

>     <Avatar src={props.avatar} alt="" />

Reviewer 2: ```suggestion
    <Avatar src={props.avatar} alt={props.username} />
```

I think our example ought to encourage users to use alt text

Author: Oh my bad. I will make this change.

Reviewer 2: No worries, just good to bake in accessibility everywhere we can!

- Use the CSS-in-JS library [Styled Components](/docs/styled-components/)
- Use [CSS Modules](/tutorial/part-two/#css-modules)

### Using Styled Components

#### Prerequisites

- A [Gatsby site](/docs/quick-start/) with an index page component

> - [gatsby-pluging-styled-components, styled-components, and babel-plugin-styled-components](/packages/gatsby-plugin-styled-components/) installed in `gatsby-config.js`

Reviewer: I know this is unusual based on the other recipes we've shipped, but I think having a step to verify these are installed correctly would help people. Maybe just show a snippet of the gatsby-config file as Step # 1?

Author: I have added it as the first step in directions.

> - Use the CSS-in-JS library [Styled Components](/docs/styled-components/)

Reviewer: We should probably remove the old recipe info, now that the Styling recipes are being added!

Author: Sorry I am not sure what should I do here. May you please elaborate?

Reviewer: Lines 378-384 can go away, as they're part of the old recipe format.

- Use [CSS Modules](/tutorial/part-two/#css-modules)

### Using Styled Components

#### Prerequisites

- A [Gatsby site](/docs/quick-start/) with an index page component

> - [gatsby-plugin-styled-components, styled-components, and babel-plugin-styled-components](/packages/gatsby-plugin-styled-components/) installed in `gatsby-config.js`

Reviewer: ```suggestion
- [gatsby-plugin-styled-components, styled-components, and babel-plugin-styled-components](/packages/gatsby-plugin-styled-components/) installed in `package.json`
```

Author: Oh this was a silly miss!
